File: compiler/bench/depth8.c
// expected: 500000000
// Stack depth 8, 500M ops
#include <stdio.h>

int main(void) {
    long stack[8];
    stack[0] = 0;  // acc
    stack[1] = 1;
    stack[2] = 2;
    stack[3] = 3;
    stack[4] = 4;
    stack[5] = 5;
    stack[6] = 6;
    stack[7] = 7;
    for (long i = 0; i < 500000000L; i++) {
        stack[0] = stack[0] + 1;
    }
    printf("%ld\n", stack[0]);
    return 0;
}

File: compiler/bench/depth8.fs
\ expected: 500000000
\ Stack depth 8, 500M ops
: main
  0 1 2 3 4 5 6 7           ( acc d1 d2 d3 d4 d5 d6 d7 )
  500000000 0 do
    >r >r >r >r >r >r >r    ( acc ) ( R: d7 d6 d5 d4 d3 d2 d1 )
    1 +                     ( acc' )
    r> r> r> r> r> r> r>    ( acc' d1 d2 d3 d4 d5 d6 d7 )
  loop
  drop drop drop drop drop drop drop . cr
;
